### PR TITLE
fastor: new package

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds `fastor`, a lightweight high performance tensor algebra framework for modern C++.

### CODE REVIEW

File: var/spack/repos/builtin/packages/fastor/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    version("0.6.4", sha256="c97a3b9dbb92413be90689af9d942cddee12a74733cf42f1a8014965553a11f8")
+
+    depends_on("cmake@3.20:", type="build")
+
+    def cmake_args(self):
+        args = [
+            self.define("BUILD_TESTING", self.run_tests),
+        ]
+        return args
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/fastor/package.py]:
```suggestion
        return [
            self.define("BUILD_TESTING", self.run_tests),
        ]
```

Optional suggestion to drop the variable declaration.

**Review by alecbcs — DISMISSED** [review]
One optional suggestion, otherwise looks good to me!

File: var/spack/repos/builtin/packages/fastor/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/fastor/package.py]: Thanks! Fixing style while I'm at it with:
```suggestion
        return [self.define("BUILD_TESTING", self.run_tests)]
```

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @wdconinc

File: var/spack/repos/builtin/packages/fastor/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/fastor/package.py]: Aside: maybe `BUILD_TESTING` should be set in `lib/spack/spack/build_systems/cmake.py` for consistency?

**Comment by wdconinc** [comment]
@spackbot run pipeline

**Pull request merged**